packages/nextjs/components/batch1/AccountInfo.tsx

```diff
@@ -0,0 +1,52 @@
+import { useAccount } from "wagmi";
+import { CheckBadgeIcon, XMarkIcon } from "@heroicons/react/24/outline";
+import { useScaffoldContractRead } from "~~/hooks/scaffold-eth";
+
+const AccountInfo = () => {
+  const accountState = useAccount();
+
+  const { data: userContractAddressData } = useScaffoldContractRead({
+    contractName: "BatchRegistry",
+    functionName: "yourContractAddress",
+    args: [accountState.address],
+  });
+
+  const { data: userAllowlistData } = useScaffoldContractRead({
+    contractName: "BatchRegistry",
+    functionName: "allowList",
+    args: [accountState.address],
+  });
+
+  return (
+    <>
+      {userAllowlistData !== undefined && userContractAddressData !== undefined && (
```

Comment:
Instead of this, you could return early (better readability)

```ts

if (!acountState.address) {
  return null;
}

return (
  <div className="py-2 px-5 ml-4 rounded-2xl text-sm font-semibold bg-[#DAE8FF]">
     .....
  </div>
)
```